deploy/infra/main.bicep

```diff
@@ -29,14 +32,46 @@ module db './webapi/db.bicep' = {
   }
 }
 
-module webApi './webapi/webapp.bicep' = {
-  name: 'webAppDeploy'
+module dbArc './webapi/dbArc.bicep' = if (customLocationId != '') {
+  name: 'dbArcDeploy'
   params: {
     location: location
+    customLocationId: customLocationId
+    dataControllerId: dataControllerId
+    subscriptionId: subscription().id
+    resourceGroupName: resourceGroup().name
+    namespace: 'appservice-ns' //Should be derived from customLocation
+    dbName: substring(uniqueString(dbName),0,5) //Name length restrictions
+    administratorLoginPassword: postgresAdminPassword
+  }
+}
+
+module webApiAzure './webapi/webappAzure.bicep' = if (customLocationId == '') {
+  name: 'webAppAzureDeploy'
+  params: {
+    location: location
+    workspaceId: monitoring.outputs.workspaceId
+    appSettingsPgHost: customLocationId == '' ? dbAzure.outputs.pgHost : dbArc.outputs.pgHost //The ternary operator is needed to ensure deployment dependencies are correct
+    appSettingsPgUser: customLocationId == '' ? dbAzure.outputs.pgUser : dbArc.outputs.pgUser //The ternary operator is needed to ensure deployment dependencies are correct
+    appSettingsPgDb: customLocationId == '' ? dbAzure.outputs.pgDb : dbArc.outputs.pgDb //The ternary operator is needed to ensure deployment dependencies are correct
```

Comment:
If we have different bicep files for arc vs regular, then why do we need this conditional?

Reply:
Bicep produces individual deployments for each module. If we don't have this ternary operator, bicep will start referencing deployments, which have not been run and the ARM deployment will fail. According to the Bicep team, this is on the backlog to be fixed. I 'think' this is the issue we're running in to, where the proposed fix is the ternary operator inside the deployment scope: https://github.com/Azure/bicep/issues/1876#issuecomment-842759172